classes/Search.php

```diff
@@ -23,7 +23,12 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  * International Registered Trademark & Property of PrestaShop SA
  */
-define('PS_SEARCH_MAX_WORD_LENGTH', 30);
+define('PS_SEARCH_MAX_WORD_LENGTH', 15);
```

Comment:
Can you make all of this constants configurable in the BO?

Reply:
Done

Reply:
I still see all `define` at the top of the file https://github.com/PrestaShop/PrestaShop/pull/14428/files#diff-4d2b8c521b121179fd17f512b9ffaebcR26 and doesn't see the Configuration values here https://github.com/PrestaShop/PrestaShop/pull/14428/files#diff-3f46cad4e139058c5f0e6f05504eab7aR143

Reply:
Right now, I put only PS_SEARCH_MAX_WORD_LENGTH as configurable variable in BO.

Would you like to have PS_SEARCH_MAX_WORDS_IN_TABLE, PS_SEARCH_ORDINATE_MIN, PS_SEARCH_ORDINATE_MAX, PS_SEARCH_ABSCISSA_MIN and PS_SEARCH_ABSCISSA_MAX as configurable values too?

I fact, I don't really advise to change this values.
To remember, in any case, the goal to change PS_SEARCH_ORDINATE_XXX or PS_SEARCH_ABSCISSA_XXX is to vary the target length of world we want to compare.

Changing :
PS_SEARCH_ORDINATE_XXX: could broke the auto scale. Only variations with +/- 0.1-0.2 could get sense for not braking anything. 
Because, if linear functions are to flat, the auto-scale is useless.
If linear functions are too leaning, the range could tends too quickly to coef max and coef min = 1. And the chance to find a matching word on small DB reduce.

A best way to do vary the slope of the straight lines softly, without braking anything, is to vary PS_SEARCH_MAX_WORDS_IN_TABLE. The result will be same, and safer.

For PS_SEARCH_ABSCISSA_XXX:
For small database, I think it's large enough (have to do more test about matching result on small DB). And for large DB, influence will not be so obvious.
Again there, if we want to decrease or increase the target length, vary PS_SEARCH_MAX_WORDS_IN_TABLE will be more conformable, and the result will be safer.

--------------------------------------------------------------------------------------------------------------

In my opinion:

Only PS_SEARCH_MAX_WORDS_IN_TABLE really matter.
- If you decrease it, you decrease the the server stress and reduce the target length
- If you increase it, you increase the the server stress and increase the target length

The only true reason to make vary this variable, could be depending the kind of activity : 
In case the shop use a wide lexical field (like book shop), by increasing PS_SEARCH_MAX_WORDS_IN_TABLE, the algo will try to target more words.

In case the shop use a small lexical field (like electronic stuff shop), by decreasing PS_SEARCH_MAX_WORDS_IN_TABLE, still  the algo will try to target less words.

--------------------------------------------------------------------------------------------------------------

Is making configurable PS_SEARCH_ORDINATE_XXX or PS_SEARCH_ABSCISSA_XXX a solution to manage stress server on small server ?
I think a better way could be to adjust PS_SEARCH_MAX_WORDS_IN_TABLE.
I can add a script to bench the server and give a value to PS_SEARCH_MAX_WORDS_IN_TABLE depending the final score.

It's not complicated, juste run a nice SQL query during installation, and give a coef to PS_SEARCH_MAX_WORDS_IN_TABLE.

--------------------------------------------------------------------------------------------------------------

What yould you like I implement ? PS_SEARCH_MAX_WORDS_IN_TABLE, PS_SEARCH_ORDINATE_MIN, PS_SEARCH_ORDINATE_MAX, PS_SEARCH_ABSCISSA_MIN and PS_SEARCH_ABSCISSA_MAX together ? Only PS_SEARCH_MAX_WORDS_IN_TABLE ?

Up to you.

Reply:
If values are not going to change, you should use class constante instead of `define` and it's ok for me :wink: